### PR TITLE
Tune AdamW baseline to 4950 steps

### PR DESCRIPTION
**Pull request by konstmish**
## Changes

- `train_steps`: 3250 -> 4950
- Replace Muon on the 2D block-matrix parameters with AdamW
- AdamW learning rates:
  - embedding: 0.7
  - projection: 0.004
  - 1D parameters: 0.015
  - 2D block matrices: 0.001
- AdamW weight decay:
  - embedding, projection, and 1D parameters: 0.001 -> 0.002
  - 2D block matrices: 1.0
- 2D block-matrix AdamW betas: `(0.9, 0.9)`
- Add a 100-step linear warmup from `1e-7`

Validation logs are included in:

`records/track_3_optimization/results/20260701_tuned_adamw_steps_4950_wd_1_wd2_0.002_matrix_b2_0.9/`

## 10-run result at 4950 steps

- mean val loss: 3.277604
- median val loss: 3.277095
- std: 0.002516
- min: 3.274590
- max: 3.282550

## README significance criterion

```text
(3.28 - mean) * sqrt(10) = 0.007577
```

This passes the required `>= 0.004`.

### CODE REVIEW

**Comment by konstmish** [comment]
I was surprised by both the tuned hyperparams and the fact that so many iterations are needed. But it appears at least to be locally optimal, combinations of learning rate and weight decay near the found values did not seem better. Warm-up is kinda optional here, it just makes the final val loss a bit better.
